### DISCUSSION TITLE
Happy to report the server working with Emacs

### DISCUSSION THREAD

**Discussion by vkazanov** [polarmutex/beancount-language-server#506] [General]
I have just setup beancount-language-server on beancount-mode / Emacs 29.1 / Ubuntu. Works like charm! Thanks a lot! Looking forward to the remaining few features, mostly definition providers.

The configration was super simple. Having beancount-language-server in PATH I added the following standard Eglot config: 

```emacs-lisp
(with-eval-after-load 'eglot
    (add-to-list 'eglot-server-programs
                 '(beancount-mode . ("beancount-language-server" "--stdio"))))
```

Probably worth adding to the README.

Thank you

**Comment by rennsax** [comment]
@polarmutex Hi! I'd like to know how to specify `journal_file` in Emacs with Eglot? Typically LSP server can recognize a directory-level configuration file but I find no documentation about this in README 😥

**Reply by dustinfarris** [reply]
fwiw, i got this working in Doom by [fixing a bug in lsp-mode](https://github.com/emacs-lsp/lsp-mode/pull/4702).

I'm not using Eglot, but it looks like you should be able to pass that in as an init option, something like:

```emacs-lisp
(with-eval-after-load 'eglot
  (add-to-list 'eglot-server-programs
               '(beancount-mode . (
                                   "beancount-language-server"
                                   "--stdio"
                                   :initializationOptions
                                   (:journal_file "/path/to/journal.beancount")))))
```
